content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js

```diff
@@ -190,11 +188,15 @@
     $(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
         selector: altInputSelector,
         validate: function() {
-            var seededValue = $(altInputSelector).attr("data-seeded-value");
-            var isAltCheckboxChecked = document.querySelector('coral-checkbox[name="./altValueFromDAM"]').checked;
+            var hasSeededValue = $(altInputSelector).val().length > 0;
+            var isImageFromPageImageChecked = document.querySelector('coral-checkbox[name="./imageFromPageImage"]').checked;
+            var isAltFromDAMChecked = document.querySelector('coral-checkbox[name="./altValueFromDAM"]').checked;
+            var isAltFromPageImageChecked = document.querySelector('coral-checkbox[name="./altValueFromPageImage"]').checked;
             var isDecorativeChecked = document.querySelector("coral-checkbox[name='./isDecorative']").checked;
             var assetWithoutDescriptionErrorMessage = "Error: Please provide an asset which has a description that can be used as alt text.";
-            if (isAltCheckboxChecked && !seededValue && !isDecorativeChecked) {
+
+            if ((isImageFromPageImageChecked && !isDecorativeChecked && !isAltFromPageImageChecked && !hasSeededValue) ||
+                (!isImageFromPageImageChecked && !isDecorativeChecked && !isAltFromDAMChecked && !hasSeededValue)) {
```

Comment:
If `!isDecorativeChecked` and `!hasSeededValue` are part of both OR statements - do you want to move them out
```suggestion
            if (((isImageFromPageImageChecked && !isAltFromPageImageChecked) || (!isImageFromPageImageChecked && !isAltFromDAMChecked))
               && !isDecorativeChecked && !hasSeededValue) {
```

Reply:
refactored while taking account the suggestion